### PR TITLE
Feature/food endpoint

### LINKED ISSUE #310: Add new Nutrition endpoint
Add a new endpoint for nutrition. We can start with fruit which I have added below.

Endpoints: 

```
/food/nutrition/fruits/
/food/nutrition/fruits/:qty
```

image

### PR DESCRIPTION
**Pull request by GauravRanganath**
## Describe your changes

Create new endpoint to generate nutritional food groups

## Link to issue this resolves

Closes #310

## Screenshot of changes

07E05C4C-7B38-4FC9-88B2-B7BC987E319C

### CODE REVIEW

**Comment by computamike** [comment]
@ageddesi  has confirmed that codacy issues can be ignored.   I'll update this branch with the latest from dev, as it currently shows as needing to be updated.

**Comment by computamike** [comment]
Just taking a look at the files in a codespace - assuming all good I intend to approve this

**Comment by ageddesi** [comment]
Can we add a comment to each of the source files please to state where the data came from, the date it was obtained and a url directing to it if has one.

**Comment by GauravRanganath** [comment]
Hey @ageddesi! Just added that information in the comments.

File: src/modules/food/tests/api/food-routes.test.ts [hunk 1]
```diff
@@ -0,0 +1,217 @@
+import app from '../../../../../app';
+let request = require('supertest');
+
+describe('food/nutrition/fruits api endpoints', () => {
+    describe('GET /food/nutrition/fruits', () => {
+        it('should return a user', async () => {
```
Comment (ageddesi) [inline src/modules/food/tests/api/food-routes.test.ts:6]: This 'it' statement is referring to a user. please change.

File: src/modules/food/tests/api/food-routes.test.ts [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+        it('should return fruits with the given quantity of fruits', async () => {
+            const response = await request(app).get(`/food/nutrition/fruits/${qty}`);
+            expect(response.body.length).toEqual(qty);
+        });
+    });
+});
+
+describe('food/nutrition/dairy api endpoints', () => {
+    describe('GET /food/nutrition/dairy', () => {
+        it('should return a user', async () => {
```
Comment (ageddesi) [inline src/modules/food/tests/api/food-routes.test.ts:38]: This 'it' statement is referring to a user. please change.

File: src/modules/food/consts/FoodEnum.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+    All,
+    Dairy,
+    Vegetables,
+    Fruits,
+    Grains,
+    Proteins,
+    Beverages,
+}
+
+export default FoodType;
```
Comment (ageddesi) [inline src/modules/food/consts/FoodEnum.ts:11]: Would be great if you can also add a test to make sure no enum types with the same name appear twice. This has caught me in the past during merges.

Comment (abhinav0022) [reply src/modules/food/consts/FoodEnum.ts:11]: Hi @ageddesi, I have made changes to the test cases and added the test case to ensure all enum types are unique. Please review them when you get a chance :)

File: src/modules/food/tests/utils/getRandomFoods.test.ts [hunk 4]
```diff
@@ -56,3 +56,11 @@ describe('get random items for all food categories', () => {
         expect(res.length).toBe(amount);
     });
 });
+
+describe('each enum type in FoodEnum must appear exactly once. FoodEnum should have unique enum types', () => {
+    it('should return the length of FoodEnum', () => {
+        const enumArray = Object.keys(FoodEnum);
+        const uniqueEnumSet = [... new Set(enumArray)];
+        expect(enumArray.length).toBe(uniqueEnumSet.length);
```
Comment (GauravRanganath) [inline src/modules/food/tests/utils/getRandomFoods.test.ts:64]: I think we can make this more concise if we did something like this`expect(new Set(Object.keys(FoodEnum))).size !== Object.keys(FoodEnum).length).toBe(true)`

Comment (abhinav0022) [reply src/modules/food/tests/utils/getRandomFoods.test.ts]: Modified the test case.

**Review by ageddesi — APPROVED** [review]

**Comment by ageddesi** [comment]
Hi @GauravRanganath 
Thank you for the contribution :)

**Pull request merged**